pkg/server/server.go

```diff
@@ -249,6 +255,36 @@ func (s *Server) Start(ctx context.Context) error {
 	}
 	logrus.WithFields(logrus.Fields{"id": s.app.ID(), "address": s.app.Address()}).Print("Started dqlite")
 
+	go func() {
+		for {
+			select {
+			case <-storageWatchTicker.C:
+				var stat unix.Statfs_t
+				unix.Statfs(s.storageDir, &stat)
```

Comment:
You could move this logic to a util as this logic is also used in the other file.
e.g. 
```
if util.diskIsFull() {
...